src/PSR7/Validators/SerializedParameter.php

```diff
@@ -125,8 +125,13 @@ private function isJsonContentType(): bool
      */
     private function castToSchemaType($value, ?string $type)
     {
-        if (($type === CebeType::BOOLEAN) && is_scalar($value) && preg_match('#^(true|false)$#i', (string) $value)) {
-            return is_string($value) ? strtolower($value) === 'true' : (bool) $value;
+        if ($type === CebeType::BOOLEAN && is_scalar($value)) {
+            if (preg_match('#^(true|false)$#i', (string)$value)) {
+                return is_string($value) ? strtolower($value) === 'true' : (bool)$value;
```

Comment:
Actually this whole check looks invalid from start. I suggest to extract `if (!is_string) ` clause as first check, returning explicit `(bool) $value` as a result and treat `$value` as a string aftert (removing all redundand checks and casts)

Reply:
Couldn't all of this be replaced with `filter_var($value, FILTER_VALIDATE_BOOLEAN)` ?

Reply:
`filter_var` allows much more words than openapi wants

https://3v4l.org/IRcHR

Reply:
Alright, fair enough. Just seems repetitive.